### PR TITLE
Supported ppc_arch as variant in env files

### PR DESCRIPTION
**Pull request by npanpaliya**
## Checklist before submitting

- [ ] Did you read the [contributor guide](https://github.com/open-ce/open-ce/blob/main/CONTRIBUTING.md)?
- [ ] Did you update any affected [documentation](https://github.com/open-ce/open-ce/blob/main/doc/)?
- [ ] Did you write any tests to validate this change?

## Description

Fixes # (issue).

## Review process to land 

1. All tests and other checks must succeed.
2. At least one [maintainer](https://github.com/open-ce/open-ce/blob/main/MAINTAINERS.md) must review and approve.
3. If any  [maintainer](https://github.com/open-ce/open-ce/blob/main/MAINTAINERS.md) requests changes, they must be addressed.

### CODE REVIEW

**Review by alexlang74 — APPROVED** [review]
This improves _what_ exactly?-)

**Comment by npanpaliya** [comment]
> This improves _what_ exactly?-)

I'm sorry for not being explanatory here.

We need to build dependencies like numpy only for P10. So, numpy feedstock should be included in open-ce env files only when `ppc_arch ==p10`. But currently our env files don't support filter like `ppc_arch`. So, to add this support, we need to make `ppc_arch` as variant while rendering the yaml files. And this PR does that exactly. We have not made `ppc_arch` as a variant all over, the way we have `build_type` or `mpi_type` or `cudatoolkit`, because it would have led many changes all over the builder plus we would also get opence conda env files generated for that ppc_arch variant too. So, to avoid these, we just added temporary variant only while loading the yaml files.

File: open_ce/conda_utils.py [hunk 1]
```diff
@@ -45,7 +45,12 @@ def render_yaml(path, variants=None, variant_config_files=None, schema=None, per
     Call conda-build's render tool to get a list of dictionaries of the
     rendered YAML file for each variant that will be built.
     """
-    config = get_or_merge_config(None, variant=variants)
+    mod_variants = {}
```
Comment (cdeepali) [inline open_ce/conda_utils.py:48]: Lets add an explanatory comment here.

**Review by cdeepali — APPROVED** [review]
LGTM.

**Pull request merged**